### DISCUSSION TITLE
'any' for return param instead of '_'?

### DISCUSSION THREAD

**Discussion by renlite** [onyx-lang/onyx#145] [Ideas]
You replaced `#aute` with `_`. 
There exists already a primitive keyword `any`. Wouldn't be `any` more meaningful?
`_` could be used for unused variables on the left.

**Comment by brendanfh** [comment]
Good question!

`any` is actually not a keyword. It is just a builtin type that is available in to all Onyx programs. You can find its definition in `core/builtin.onyx` and it will look something like this:

```fs
any :: struct {
    data: rawptr
    type: type_expr
}
```

`any` is used where there is a *runtime-known* dynamic return type. When writing a function that returns `any`, you have to allocate a buffer for the data of the `any`, and the return an instance of the `any` structure. In other words, `any` is just like other structures; it just has special syntax highlighting and a name akin to the idea of `any` in TypeScript or similar languages.

I like to think of the `_` as a "hole" that the compiler will fill. It is also allowed as the type of a local variable, that the compiler will figure out when the variable is first assigned to.

```fs
main :: () {
    x: _
    x = 10
}
```
